src/btrfs.cpp

```diff
@@ -0,0 +1,154 @@
+/****************************************************************************************
+**
+** Copyright (C) 2015 Jolla Ltd.
+** Contact: Martin Grimme <martin.grimme@gmail.com>
+** All rights reserved.
+**
+** This file is part of the btrfs-balancer package.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+
+#include "btrfs.h"
+
+#include <QLocale>
+#include <QRegExp>
+#include <QDebug>
+
+namespace
+{
+// path of the btrfs tool
+const QString BTRFS_PATH("/usr/sbin/btrfs");
+
+// default root device to use
+const QString ROOT_DEVICE("/dev/mmcblk0p28");
+
+// regexp to retrieve size usage information from btrfs output
+const QRegExp RE_USAGE("size ([0-9.]+\\w+) used ([0-9.]+\\w+)");
+
+qint64 parseSize(const QString& size)
+{
+    double value = -1;
+
+    QLocale defaultLocale = QLocale();
+    QLocale::setDefault(QLocale::C);
```

Comment:
Looks unnecessary, QString: "The string conversion will always happen in the 'C' locale. For locale dependent conversion use QLocale::toDouble()"